### PR TITLE
Add TC-DRLK-2.12

### PR DESCRIPTION
**Pull request by mmarc**
Adds TC-DRLK-2.12 for Unbolt feature introduced with https://github.com/CHIP-Specifications/chip-test-plans/pull/2641

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

File: src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+
+    - label: "TH reads the RequirePINforRemoteOperation attribute from the DUT"
+      PICS: DRLK.S.F07 && DRLK.S.F00 && DRLK.S.A0033
+      command: "readAttribute"
+      attribute: "RequirePINforRemoteOperation"
+      response:
+          value: false
+
+    - label: "TH sends the Unbolt Door command to the DUT without PINCode"
+      PICS: DRLK.S.C27.Rsp
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml:158]: Test plan has his conditioned on "DRLK.S.C01.Rsp(Unlock Door)" (which is wrong).  Is there a test plan issue tracking this?

File: src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+      command: "UnboltDoor"
+      timedInteractionTimeoutMs: 1000
+
+    - label: "TH sends the Unbolt Door command to the DUT with valid PINCode"
+      PICS: DRLK.S.C27.Rsp
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml:163]: Again, wrong PICS in test plan.

File: src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+          error: UNSUPPORTED_WRITE
+
+    - label: "TH reads the RequirePINforRemoteOperation attribute from the DUT"
+      PICS: DRLK.S.F07 && DRLK.S.F00 && DRLK.S.A0033
+      command: "readAttribute"
+      attribute: "RequirePINforRemoteOperation"
+      response:
+          value: true
+
+    - label: "TH sends the Unbolt Door command to the DUT with valid PINCode"
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml:198]: Label does not match test plan, but here the test plan is wrong.  Is that tracked?

File: src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml [hunk 4]
```diff
@@ -0,0 +1,435 @@
...
+      PICS: DRLK.S.F00 && DRLK.S.F01 && DRLK.S.A0030
+      command: "writeAttribute"
+      attribute: "WrongCodeEntryLimit"
+      arguments:
+          value: 3
+
+    - label:
+          "TH writes WrongCodeEntryLimit attribute value as 3 on the DUT and
+          verify DUT responds with UNSUPPORTED_WRITE"
+      PICS: DRLK.S.F00 && DRLK.S.F01 && ! DRLK.S.A0030
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml:237]: This does not look like the right PICS check?  The test plan is checking `DRLK.S.A0030.Write`, afaict.  Same in the previous step.

File: src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml [hunk 5]
```diff
@@ -0,0 +1,435 @@
...
+      PICS: DRLK.S.F00 && DRLK.S.F01 && DRLK.S.A0031
+      command: "writeAttribute"
+      attribute: "UserCodeTemporaryDisableTime"
+      arguments:
+          value: 15
+
+    - label:
+          "TH writes UserCodeTemporaryDisableTime attribute value as 15 Seconds
+          on the DUT and Verify that the DUT sends Success response"
+      PICS: DRLK.S.F00 && DRLK.S.F01 && ! DRLK.S.A0031
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml:257]: Again, this does not look like the right PICS check.

File: src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml [hunk 6]
```diff
@@ -0,0 +1,435 @@
...
+      command: "UnboltDoor"
+      timedInteractionTimeoutMs: 1000
+      arguments:
+          values:
+              - name: "PINCode"
+                value: "1234568"
+      response:
+          error: FAILURE
+
+    - label: "TH sends the Unbolt Door command to the DUT with invalid PINCode"
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml:298]: Test plan has 3 invalid PIN unbolts.  Why does this test have 4?

File: src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml [hunk 7]
```diff
@@ -0,0 +1,435 @@
...
+                value: "1234568"
+      response:
+          error: FAILURE
+
+    - label:
+          "TH reads the UserCodeTemporaryDisableTime attribute from the DUT and
+          check attribute is triggered"
+      PICS: DRLK.S.F00 && DRLK.S.F01 && DRLK.S.A0031
+      command: "readAttribute"
+      attribute: "UserCodeTemporaryDisableTime"
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml:314]: Reading this attribute does not check what the test plan says to check.  Why is there a read here at all?

File: src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml [hunk 8]
```diff
@@ -0,0 +1,435 @@
...
+      cluster: "DelayCommands"
+      command: "WaitForMs"
+      PICS: DRLK.S.F00 && DRLK.S.F01 && DRLK.S.A0031
+      arguments:
+          values:
+              - name: "ms"
+                value: 15000
+
+    - label: "TH writes AutoRelockTime attribute value as 10 seconds on the DUT"
+      PICS: DRLK.S.A0023 && PICS_SDK_CI_ONLY
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml:339]: If we're doing CI, why not just use 2-3 seconds instead of 10?

**Review by bzbarsky-apple — COMMENTED** [review]
The test plan really needs to be testing something about events and how Unbolt should be different from Unlock, no?

File: src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml (hunk 3 above)
Comment (mmarc) [reply src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml:198]: https://github.com/CHIP-Specifications/chip-test-plans/pull/2801

File: src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml (hunk 2 above)
Comment (mmarc) [reply src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml:163]: https://github.com/CHIP-Specifications/chip-test-plans/pull/2801

File: src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml (hunk 1 above)
Comment (mmarc) [reply src/app/tests/suites/certification/Test_TC_DRLK_2_12.yaml:158]: https://github.com/CHIP-Specifications/chip-test-plans/pull/2801

**Review by andy31415 — DISMISSED** [review]

**Comment by mmarc** [comment]
@bzbarsky-apple 
> The test plan really needs to be testing something about events and how Unbolt should be different from Unlock, no?

The events are handled in TC-DRLK-2.10 (https://github.com/CHIP-Specifications/chip-test-plans/blob/master/src/cluster/door_lock_Cluster.adoc#tc-drlk-2-10-verification-for-the-door-lock-events-dut-server)

**Comment by mmarc** [comment]
This PR is now obsolete as TC-DRLK-2.12 has been added with https://github.com/project-chip/connectedhomeip/pull/27345

**Pull request closed without merging**